xstream/src/java/com/thoughtworks/xstream/XStream.java

```diff
@@ -996,6 +998,7 @@ protected void setupConverters() {
         registerConverter(new JavaMethodConverter(classLoaderReference), PRIORITY_NORMAL);
         registerConverter(new JavaFieldConverter(classLoaderReference), PRIORITY_NORMAL);
         registerConverter(new LambdaConverter(mapper, reflectionProvider, classLoaderReference), PRIORITY_NORMAL);
+        registerConverter(new AtomicBooleanConverter(), PRIORITY_NORMAL);
```

Comment:
You should also register an alias "atomic-boolean".

Reply:
That doesn't fit with my design, which is to re-implement this functionality without reflection without changing the on-disk format.